horovod/mxnet/__init__.py

```diff
@@ -64,16 +66,23 @@ def _do_allreduce(self, index, grad):
                 index_split = split_list(index, self._num_groups)
 
                 for i, (grads, indices) in enumerate(zip(grad_split, index_split)):
-                    grouped_allreduce_(tensors=grads, average=False, name="{}:{}".format(indices[0], indices[-1]), priority=-i,
+                    tensors_compressed, ctxs = zip(*[self._compression.compress(grad) for grad in grads])
+                    grouped_allreduce_(tensors=tensors_compressed, average=False,
+                                       name="{}:{}".format(names[0], names[-1]), priority=-i,
                                        prescale_factor=1.0 / self._gradient_predivide_factor)
+                    grads = [self._compression.decompress(t, ctx) for t, ctx in zip(tensors_compressed, ctxs)]
             else:
               for i in range(len(index)):
-                  allreduce_(grad[i], average=False,
+                  tensor_compressed, ctx = self._compression.compress(grad[i])
+                  allreduce_(tensor_compressed, average=False,
                              name=str(index[i]), priority=-i,
                              prescale_factor=1.0 / self._gradient_predivide_factor)
+                  grad[i] = self._compression.decompress(tensor_compressed, ctx)
```

Comment:
This line errors out as `grad` is a tuple so you can't assign the resulting tensor to `grad[i]`. `grad[i][:]` might work but the copy is not ideal.